tests-and-examples/tests/cases/misc.rs

```diff
@@ -221,3 +221,42 @@ fn test_shutdown_when_exists_grpc_call() {
         "Send should get error because server is shutdown, so the grpc is cancelled."
     );
 }
+
+#[test]
+fn test_custom_checker_server_side() {
+    let flag_1 = Arc::new(atomic::AtomicBool::new(false));
+    let flag_2 = flag_1.clone();
+
+    let env = Arc::new(Environment::new(2));
+    // Start a server and delay the process of grpc server.
+    let service = create_greeter(PeerService);
+    let mut server = ServerBuilder::new(env.clone())
+        .add_checker(move |ctx| {
+            let method = String::from_utf8(ctx.method().to_owned());
+            assert_eq!(&method.unwrap(), "/helloworld.Greeter/SayHello");
+
+            if flag_1.load(Ordering::Relaxed) {
+                CheckResult::Abort(RpcStatus::new(RpcStatusCode::DATA_LOSS, None))
+            } else {
+                CheckResult::Continue
+            }
+        })
+        .register_service(service)
+        .bind("127.0.0.1", 0)
+        .build()
+        .unwrap();
+    server.start();
+    let port = server.bind_addrs().next().unwrap().1;
+    let ch = ChannelBuilder::new(env).connect(&format!("127.0.0.1:{}", port));
+    let client = GreeterClient::new(ch);
+    let req = HelloRequest::default();
+
+    let _ = client.say_hello(&req).unwrap();
+    let _ = client.say_hello(&req).unwrap();
+
+    flag_2.store(true, Ordering::Relaxed);
```

Comment:
should use SeqCst order for both of the atomic operations.

Reply:
Address comment, BTW any ordering used here is correct.

Reply:
My mistake, if not use `SeqSct`, `flag_2.store(true, Ordering::Relaxed);` maybe be reordered behind assert.